src/main/java/io/supertokens/inmemorydb/Start.java

```diff
@@ -847,4 +857,324 @@ public void setJWTSigningKey_Transaction(TransactionConnection con, JWTSigningKe
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+
+    }
+
+    @Override
+    public PasswordlessCode[] getCodesOfDevice_Transaction(TransactionConnection con, String deviceIdHash)
+            throws StorageQueryException {
+        Connection sqlCon = (Connection) con.getConnection();
+        try {
+            return PasswordlessQueries.getCodesOfDevice_Transaction(this, sqlCon, deviceIdHash);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public void deleteDevice_Transaction(TransactionConnection con, String deviceIdHash) throws StorageQueryException {
+        Connection sqlCon = (Connection) con.getConnection();
+        try {
+            PasswordlessQueries.deleteDevice_Transaction(this, sqlCon, deviceIdHash);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+
+    }
+
+    @Override
+    public void deleteDevicesByPhoneNumber_Transaction(TransactionConnection con, @Nonnull String phoneNumber)
+            throws StorageQueryException {
+        Connection sqlCon = (Connection) con.getConnection();
+        try {
+            PasswordlessQueries.deleteDevicesByPhoneNumber_Transaction(this, sqlCon, phoneNumber);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public void deleteDevicesByEmail_Transaction(TransactionConnection con, @Nonnull String email)
+            throws StorageQueryException {
+        Connection sqlCon = (Connection) con.getConnection();
+        try {
+            PasswordlessQueries.deleteDevicesByEmail_Transaction(this, sqlCon, email);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public void createCode(String codeId, String deviceIdHash, String linkCodeHash, long createdAt)
+            throws StorageQueryException, UnknownDeviceIdHash, DuplicateCodeIdException,
+            DuplicateLinkCodeHashException {
+
+        try {
+            this.startTransaction(con -> {
+                Connection sqlCon = (Connection) con.getConnection();
+                // SQLite is not compiled with foreign key constraint and so we must check for
+                // the deviceIdHash manually
+                if (this.getDevice_Transaction(con, deviceIdHash) == null) {
+                    throw new StorageTransactionLogicException(new UnknownDeviceIdHash());
+                }
+                try {
+                    PasswordlessQueries.createCode_Transaction(this, sqlCon, codeId, deviceIdHash, linkCodeHash,
+                            createdAt);
+                    sqlCon.commit();
+                } catch (SQLException e) {
+                    throw new StorageTransactionLogicException(e);
+                }
+                return null;
+            });
```

Comment:
I prefer if this is moved to the PasswordlessQueries.java file inside a function named `createCode` so that the contents of this file can be somewhat closer to that of mysql's and posthresql's